### PR TITLE
🔔 Add Firebase push notifications service

### PR DESCRIPTION
**Pull request by jurgenwerk**
Ticket: [CS-2701](https://linear.app/cardstack/issue/CS-2701/add-firebase-cloud-messaging-service)

This PR adds the Firebase client which the notification worker will use to send push notifications. 

We'll have to align with the mobile team to define the app link so that the user will get to the relevant screen when tapping the notification.

There are couple of new ENV vars: FIREBASE_PROJECT_ID, FIREBASE_CLIENT_EMAIL, FIREBASE_PRIVATE_KEY, FIREBASE_DATABASE_URL which should be added in .env for development. These can be obtained by going to [https://console.firebase.google.com](https://console.firebase.google.com), choosing Project Settings > Service accounts > Generate new private key.

### CODE REVIEW

File: packages/hub/services/push-notifications/firebase.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+interface FCMPayload {
+  message: {
+    notification: {
+      body: string;
+      title: string;
+      data: any; // TODO: Define deep link
+    };
+  };
+  token: string;
```
Comment (jurgenwerk) [inline packages/hub/services/push-notifications/firebase.ts:11]: This is the push client id

File: packages/hub/services/push-notifications/firebase.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+}
+
+export default class FirebasePushNotifications {
+  async send(payload: FCMPayload) {
+    await admin.messaging().send(payload);
```
Comment (jurgenwerk) [inline packages/hub/services/push-notifications/firebase.ts:16]: Error handling is missing here. But I think we can add this later when we start testing and see the possible edge cases

**Review by backspace — APPROVED** [review]
agreed re waiting to see how things actually unfold when it’s live 👍🏻😀
